lib/jobs/install-os.js

```diff
@@ -310,10 +280,14 @@ function installOsJobFactory(
             self._subscribeNodeNotification(self.nodeId, function(data) {
                 assert.object(data);
 
-                return self.updateGraphProgress(self.options.progressMilestones.completed)
-                .then(function() {
-                    return self._done();
-                });
+                if(_.has(self.options, 'progressMilestones.completed')) {
+                    graphProgressService.publishTaskProgress(
+                        self.context.graphId,
+                        self.taskId,
+                        self.options.progressMilestones.completed
+                    );
+                }
+                return self._done();
```

Comment:
Since `graphProgressService.publishTaskProgress` is a promise, so the `_done` is called before the publishTaskProgress finishes, this means the job is finished with some background code is still in execution.

Please change this to make sure the _done is called when everything is OK.